src/vs/editor/contrib/smartSelect/common/smartSelect.ts

```diff
@@ -196,3 +198,84 @@ class ShrinkSelectionAction extends AbstractSmartSelect {
 		});
 	}
 }
+
+@editorAction
+class SelectBracketAction extends EditorAction {
+	constructor() {
+		super({
+			id: 'editor.action.expandSelectionToBrackets',
+			label: nls.localize('smartSelect.expandSelectionToBrackets', "Expand Selection To Brackets"),
+			alias: 'Expand Selection To Brackets',
+			precondition: null,
+			kbOpts: {
+				kbExpr: EditorContextKeys.textFocus,
+				primary: KeyMod.CtrlCmd | KeyMod.Shift | KeyCode.US_SLASH
+			}
+		});
+	}
+
+	public run(accessor: ServicesAccessor, editor: ICommonCodeEditor): void {
+		const model = editor.getModel();
+		if (!model) {
+			return;
+		}
+
+		const selection = editor.getSelection();
+		if (!selection.isEmpty()) {
+			return;
+		}
+
+		const originalPosition = selection.getStartPosition();
+
+		// find previous open bracket with no match
+		let prevBracket = null;
+		let currentPosition = new Position(originalPosition.lineNumber, originalPosition.column);
+		let setCount = 0;
+		do {
+			prevBracket = model.findPrevBracket(currentPosition);
+			if (prevBracket) {
+				currentPosition = new Position(prevBracket.range.startLineNumber, prevBracket.range.startColumn);
+				!prevBracket.isOpen ? setCount++ : setCount--;
+			}
+		} while (prevBracket && !(prevBracket.isOpen && setCount < 0));
+
+		// find next closed bracket with no match
+		let nextBracket = null;
+		currentPosition = new Position(originalPosition.lineNumber, originalPosition.column);
+		setCount = 0;
+		do {
+			nextBracket = model.findNextBracket(currentPosition);
+			if (nextBracket) {
+				currentPosition = new Position(nextBracket.range.endLineNumber, nextBracket.range.endColumn);
+				nextBracket.isOpen ? setCount++ : setCount--;
+			}
+		} while (nextBracket && (nextBracket.isOpen && setCount < 0));
+
+		// check validity of prev and next brackets
+		if (!prevBracket && !nextBracket) {
+			return;
+		}
+
+		// pick the outermost set of brackets
+		const matchPrev = prevBracket ? model.matchBracket(new Position(prevBracket.range.startLineNumber, prevBracket.range.startColumn)) : null;
+		const matchNext = nextBracket ? model.matchBracket(new Position(nextBracket.range.startLineNumber, nextBracket.range.startColumn)) : null;
+		let bracketsToSelect = null;
+		if (!matchPrev) {
+			bracketsToSelect = matchNext;
+		} else if (!matchNext) {
+			bracketsToSelect = matchPrev;
+		} else if (!new Position(matchPrev[1].endLineNumber, matchPrev[1].endColumn)
```

Comment:
Comparing line/column is better than creating object and relying on their comparable methods.